DeliveryServiceEF/DeliveryServiceEF.Data/UnitOfWork.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DeliveryServiceEF.Domain.Interfaces;
+
+namespace DeliveryServiceEF.Data
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        private readonly DataContext _context;
+        public ICategoryRepository Categories { get; }
+        public IOrderRepository Orders { get; }
+        public IProductRepository Products { get; }
+        public IUserRepository Users { get; }
+
+        public UnitOfWork(DataContext context)
+        {
+            _context = context;
+            Categories = new CategoryRepository(_context);
```

Comment:
seems like u can remove "CategoryRepository" and use "Repository<Category>"
as well as for other repos

Reply:
and think to inject them via contructors as for context